gpii/node_modules/displaySettingsHandler/src/displaySettingsHandler.js

```
@@ -0,0 +1,211 @@
+    //      DWORD dmDisplayOrientation;
+    //      DWORD dmDisplayFixedOutput;
+    //  };
+    //};
+    ["short", "dmColor"],
+    ["short", "dmDuplex"],
+    ["short", "dmYResolution"],
+    ["short", "dmTTOption"],
+    ["short", "dmCollate"],
+    [arrayType(t.TCHAR, CCHFORMNAME), "dmFormName"],
+    [t.WORD, "dmLogPixels"],
+    [t.DWORD, "dmBitsPerPel"],
+    [t.DWORD, "dmPelsWidth"],
+    [t.DWORD, "dmPelsHeight"],
+    //union {
+    [t.DWORD, "dmDisplayFlags"],
+    //  DWORD dmNup;
+    //};
+    [t.DWORD, "dmDisplayFrequency"],
+    //#if (WINVER >= 0x0400)
+    [t.DWORD, "dmICMMethod"],
+    [t.DWORD, "dmICMIntent"],
+    [t.DWORD, "dmMediaType"],
+    [t.DWORD, "dmDitherType"],
+    [t.DWORD, "dmReserved1"],
+    [t.DWORD, "dmReserved2"],
+    //#if (WINVER >= 0x0500) || (_WIN32_WINNT >= 0x0400)
+    [t.DWORD, "dmPanningWidth"],
+    [t.DWORD, "dmPanningHeight"]
+    //#endif
+    //#endif
+]);
+
+windows.display.user32 = ffi.Library("user32", {
+    // https://msdn.microsoft.com/en-us/library/windows/desktop/dd162611(v=vs.85).aspx
+    // LPCWSTR, DWORD, DEVMODE*
+    "EnumDisplaySettingsW": [
+        t.BOOL, ["pointer", t.DWORD, "pointer"]
+    ],
+    // https://msdn.microsoft.com/en-us/library/windows/desktop/dd183411(v=vs.85).aspx
+    // DEVMODE*, DWORD
+    "ChangeDisplaySettingsW": [
+        t.LONG, ["pointer", t.DWORD]
+    ]
+});
+
+/**
+    *  Gets the current display's screen resolution
+    *
+    * @return {Object) The width and height of the screen.
+    */
+windows.display.getScreenResolution = function () {
+    var dm = new windows.DEVMODEW();
+    dm.ref().fill(0);
+    dm.dmSize = windows.DEVMODEW.size;
+
+    if (c.FALSE !== windows.display.user32.EnumDisplaySettingsW(ref.NULL, c.ENUM_CURRENT_SETTINGS, dm.ref())) {
+        // note for unknown reason on win 10 the returned dmSize is 188 not expected 220
+        return { width: dm.dmPelsWidth, height: dm.dmPelsHeight };
+    }
+    return { width: 0, height: 0 };
```

Well, I'd say that we should raise an error if we can't get the screen resolution. I've changed this to raise an error if we can't get the screen resolution so the system can take this into account. Let me know if this makes sense to you.